controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -73,6 +73,106 @@ static bool validate_hex_notation(const std::string &str, uint8_t expected_octet
     return true;
 }
 
+/**
+ * @brief Parse bss_info string into bss_info_conf_struct.
+ * 
+ * @param[in] bss_info_str String conataining bss info configuration
+ * @param[out] bss_info_conf Controller database struct filled with configuration.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if successful, false if not.
+ */
+static std::string parse_bss_info(const std::string &bss_info_str,
+                                  db::bss_info_conf_t &bss_info_conf, std::string &err_string)
+{
+    auto confs = string_utils::str_split(bss_info_str, ' ');
+    if (confs.size() != 8) {
+        err_string = "missing configuration";
+        return std::string();
+    }
+
+    // Alid
+    std::string al_mac(confs[0]);
+    if (!net::network_utils::is_valid_mac(al_mac)) {
+        err_string = "invalid al_mac";
+        return std::string();
+    }
+
+    bss_info_conf = {};
+
+    // Operating class
+    uint operating_class = string_utils::stoi(confs[1]);
+    if (operating_class > UINT8_MAX || operating_class == 0) {
+        err_string = "invalid operating class";
+        return std::string();
+    }
+    bss_info_conf.operating_class = operating_class;
+
+    // SSID
+    bss_info_conf.ssid = confs[2];
+    if (bss_info_conf.ssid.size() > WSC::eWscLengths::WSC_MAX_SSID_LENGTH) {
+        err_string = "ssid is too long";
+        return std::string();
+    }
+
+    // Authentication type
+    auto &authentication_type_str = confs[3];
+    if (!validate_hex_notation(authentication_type_str, 2)) {
+        err_string = "invalid authentication type format";
```

Comment:
Wouldn't it be useful to also include the value in the error string? I.e. `err_string = "..." + authentication_type_str`

Reply:
Maybe, but from the outside, I add to which bss_info this failed, so the user can easily locate its problem.